src/suggestion.rs

```diff
@@ -50,14 +69,206 @@ impl fmt::Display for Detector {
     }
 }
 
+// For long lines, literal will be trimmed to display in one terminal line.
+// Misspelled words that are too long shall also be ellipsized.
+pub fn condition_display_content(
+    terminal_size: usize,
+    _indent: usize,
+    stripped_line: &str,
+    mistake_range: Range,
+    padding_till_excerpt_start: usize,
+    marker_size: usize,
+) -> (String, usize, usize) {
+    if stripped_line.chars().count() + padding_till_excerpt_start <= terminal_size {
+        return (stripped_line.to_owned(), mistake_range.start, marker_size);
+    }
+
+    // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+    //
+    //|---------------------------------------------------------------------------------------| terminal_size
+    //|-------| padding_till_excerpt_start = indent (3+line_number_digit_count) + 2 white spaces = 7usize, for this case.
+    //     |------| offset = PADDING_OFFSET; 3 chars for `...` and 2 white spaces more added in the formatting.
+    //
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+    const MAX_MISTAKE_LEN: usize = 20;
+
+    const HEAD_DISPLAY_LEN: usize = 4;
+    const TAIL_DISPLAY_LEN: usize = 4;
+    const DOTS_LEN: usize = 3;
+
+    assert!(HEAD_DISPLAY_LEN + DOTS_LEN + TAIL_DISPLAY_LEN <= MAX_MISTAKE_LEN);
+
+    const TOTAL_CONTEXT_CHAR_COUNT: usize = 10;
```

Comment:
it can be a bit confusing because there is a chunk and there is one line in the chunk with a mistaken word, online that is too long. I would be confused with which context this is referring to because `context` has a meaning of words or some text but this is just spaces or paddings.

Reply:
Good point, I'll reword this

Reply:
I couldn't really come up with a better wording.. I think I'll leave it for now as is, once this is merged I'd be happy to accept a new PR improving the wording